viewer/src/utilities/RxCounter.ts

```diff
@@ -48,20 +53,29 @@ export class RxCounter {
    * Get counter observable.
    * @return {Observable<number>} The counter observable.
    */
-  public countObservable(): Observable<number> {
+  public progressObservable(): Observable<LoadingState> {
     return this._actionSubject.pipe(
-      scan((count, action) => {
-        switch (action) {
-          case Action.Increment:
-            return count + 1;
-          case Action.Decrement:
-            return count - 1;
-          case Action.Reset:
-            return 0;
-          default:
-            throw new Error(`Unsupported action ${action}`);
-        }
-      }, 0),
+      scan(
+        (loadingState, action) => {
+          switch (action) {
+            case Action.Increment:
+              loadingState.itemsRequested++;
+              break;
+            case Action.Decrement:
+              loadingState.itemsLoaded++;
+              break;
+            case Action.Reset:
+              loadingState.itemsLoaded = 0;
+              loadingState.itemsRequested = 0;
+              break;
+            default:
+              throw new Error(`Unsupported action ${action}`);
+          }
+          return loadingState;
+        },
+        { itemsLoaded: 0, itemsRequested: 0 } as LoadingState
+      ),
+      throttleTime(200, asyncScheduler, { trailing: true }),
```

Comment:
I changed it in a commit I did from home, but since I was running for the bus it seems it failed and I didn't catch it. Wasn't a lot of change, and I remember what I did so I'll sort it out quickly. The throttle is not needed there, but it is nice to have a throttle, since say the count goes up super fast, you don't need all the emissions, what you are really interested in is the last emission, and not necessarily all the emissions.